test-data/unit/check-enum.test

```diff
@@ -6,10 +6,19 @@ class Medal(Enum):
     gold = 1
     silver = 2
     bronze = 3
+reveal_type(Medal.bronze)  # E: Revealed type is '__main__.Medal'
 m = Medal.gold
-m = 1
-[out]
-main:7: error: Incompatible types in assignment (expression has type "int", variable has type "Medal")
+m = 1  # E: Incompatible types in assignment (expression has type "int", variable has type "Medal")
+
+[case testEnumFromEnumMetaBasics]
```

Comment:
I think it would be helpful to have a testcase for something that inherits from a class that has `EnumMeta` as a metaclass, to check those classes are still found to be `Enum`s for all intents and purposes.

Also perhaps a test of a `EnumMeta` deriving class that tries to use generics?

Reply:
>I think it would be helpful to have a testcase for something that inherits from a class that has EnumMeta as a metaclass, to check those classes are still found to be Enums for all intents and purposes.

Will do, but note that this is already handled by classes that inherit `Enum` itself.

I'm not sure I understand the second suggestion. Should mypy support something like that?

Reply:
Good point on the first, but just to be safe. Also on the second, I mean to check that it fails correctly :)